digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/MunicipalityApi.scala

```diff
@@ -1,1132 +1,390 @@
-        val assetTypeId = getAssetTypeId(assetTypeName)
-        AssetTypeInfo.apply(assetTypeId).geometryType match {
-          case "linear" =>
-            if(isDynamic(assetTypeId)) {
-              dynamicPieceWiseAssetsToApi(getLinearAssetsByMunicipality(municipalityCode, assetTypeId))
-            } else {
-              linearPieceWiseAssetsToApi(getLinearAssetsByMunicipality(municipalityCode, assetTypeId))
-            }
-          case "point" => pointAssetsToApi(getPointAssetsByMunicipality(municipalityCode, assetTypeId), assetTypeId)
-          case _ =>
-        }
-    }
-  }
+  def getFeatureErrorsByDatasetId(datasetId: String, datasetFeaturesWithoutIds: Int): Any = {
+    val featuresStatusCode = awsDao.getAllFeatureIdAndStatusByDataset(datasetId).filter { case (_, status) => status != DatasetStatus.ErrorsFeatures.value.toString}
 
-  private def isDynamic(assetId: Int): Boolean = {
-    val dynamicAssetType = Seq(PavedRoad.typeId, MassTransitLane.typeId)
-
-    dynamicAssetType.contains(assetId)
-  }
-
-  //Description of Api entry point to get assets by id
-  val getAssetById =
-    (apiOperation[ExampleItemResponseOnGet]("getAssetById")
-      .parameters(
-        pathParam[String]("assetId").description("Id of the asset to be fetch"),
-        pathParam[String]("assetType").description("Asset type name to get")
-      )
-      .responseMessages(
-        ResponseMessage(401, "The authentication parameters are missing or are incorrect"),
-        ResponseMessage(403, "The authenticated user doesn't have permissions to do this operation on that municipality."),
-        ResponseMessage(404, "Municipality code or asset type not found.")
+    val featuresStatusMap = featuresStatusCode.map(tuple =>
+      Map(
+        "FeatureId" -> tuple._1.toString,
+        "Message" -> tuple._2.split(",").map(message => FeatureStatus(message.toInt).description)
       )
-      tags "Municipality API"
-      summary "Returns information about a single asset ."
-      authorizations "Contact your service provider for more information"
-      description
-      "Example URL: /api/municipality/lighting/400018"
-      )
-
-  get("/:assetType/:assetId", operation(getAssetById)) {
-    contentType = formats("json")
-    val assetId = params("assetId").toInt
-    val assetTypeName = params("assetType")
-
-    assetTypeName match {
-      case "manoeuvre" =>
-        manoeuvreAssetToApi(getManoeuvreAndRoadLinks(Seq(assetId)).headOption.getOrElse(halt(NotFound("Asset not found"))))
-      case "speed_limit" =>
-        speedLimitAssetToApi(getSpeedLimitsAndRoadLinks(Set(assetId)).headOption.getOrElse(halt(NotFound("Asset not found"))))
-      case _ =>
-        val assetTypeId = getAssetTypeId(assetTypeName)
-        AssetTypeInfo.apply(assetTypeId).geometryType match {
-          case "linear" =>
-            if(isDynamic(assetTypeId)) {
-              dynamicAssetToApi(getLinearAssetsAndRoadLinks(assetTypeId, Set(assetId)).headOption.g
```

Comment:
`` scala
) ++ { if (datasetFeaturesWithoutIds != 0) { Map("Features without ids" -> datasetFeaturesWithoutIds.toString) }
``

Reply:
Tried it and it does not seems to work since I need to always return something